joingroup.go

```diff
@@ -0,0 +1,212 @@
+package kafka
+
+import (
+	"bufio"
+	"bytes"
+)
+
+type memberProtocolMetadata struct {
+	// MemberID assigned by the group coordinator or null if joining for the
```

Comment:
Is `null` the `"null"` string here? Or the zero-value of a `string`?

Reply:
It's the zero-value of a string.  I can update the docs to clarify.